[ENH] BEP036 - Phenotypic Data Guidelines (#2123)
bids-standard/bids-specification: src/appendices/phenotype.md

| `acq_time`       | OPTIONAL        | If acquisition time is available, the `acq_time` column CAN be used to record the time of acquisition of each row in the tabular file. |

Furthermore, if you have to add a `session_id` column to the
tabular phenotypic data, you then MUST also introduce a session directory to the
imaging data, even if only one imaging session has been created.
This rule can be considered as "**if anyone uses sessions, everyone uses sessions**."
And vice versa, if imaging data has session directories,
all imaging data and tabular phenotypic data MUST have sessions.

This produces a file in which same-participant entries can take up as many rows
as needed according to the smallest unit of acquisition.
The combination of values in the `participant_id`, `session_id`, and `run` (if present)
columns MUST be unique for the entire tabular file.

### 4. Add `MeasurementToolMetadata` to each tabular phenotypic measurment tool

Whenever possible, it is RECOMMENDED to add `MeasurementToolMetadata` to
each `phenotype/<measurement_tool_name>.json` data dictionary.
This improves reusability and provides clarity about the measurement tool.

> ### 5. Use the demographics file for common variables about participants

Author: @VisLab Great to hear from you! Thanks for reading the BEP. Now to your comments...

> Example 1: A dataset in which there are multiple training sessions on some equipment or on some test procedure on multiple different days before any imaging or any data that would have a run occurs.

We may not have made it explicit, but [here is this example](https://bids-specification--2123.org.readthedocs.build/en/2123/appendices/phenotype.html#1-participant-with-2-sessions-where-1-session-is-only-tabular-phenotype-and-the-other-is-only-imaging) in the HTML preview of the appendix. It covers what to do when there's "1 participant with 2 sessions, where 1 session is only tabular phenotype and the other is only imaging". The same idea extends to more than 2 sessions of some combination of acquired data and tabular phenotypic data. For instance, in your Example 1, we could say:

### Define many phenotypic session names

- `ses-day1equipmenttraining1`
- `ses-day1equipmenttraining2`
- `ses-day1equipmenttraining3`
- `ses-day2equipmenttraining1`
- `ses-day2equipmenttraining2`
- `ses-preproceduretraining`
- `ses-procedure`
- `ses-postprocedure`
- ...

### Store most, if not all, in one aggregated file

- `phenotype/equipment_training.tsv`
  - Where `participant_id`, `session_id`, `run_id`, and `day` are joint indexes.
  - Yes, `day` is not a a validate-able column, but you could alternatively create a `phenotype/day#_equipment_training.tsv` if you want stricter validation support.
  - Remember that `run_id` is not about tying the data to a particular task's run, but instead to differentiate multiple runs of the same tabular phenotypic test, survey, assessment, questionnaire, or lab result within the same `participant_id`'s `session_id`.

> Example 2: Sleep questionnaires and other questionnaires that administered multiple times un-associated with any imaging sessions.

The answer to example 2 is [the appendix example I linked above, and now here](https://bids-specification--2123.org.readthedocs.build/en/2123/appendices/phenotype.html#1-participant-with-2-sessions-where-1-session-is-only-tabular-phenotype-and-the-other-is-only-imaging).

I hope this helps and let me know if you don't think this resolves your comment. Or if perhaps there's something else we should consider which you can propose?

> ### 5. Use the demographics file for common variables about participants

Author: @timdressler

> I want to draw attention to a potential issue with the handling repeated usage of a phenotype/ measurement. Currently, if a measurement is conducted more than once, a run_id column MUST be added that "corresponds to an existing run- entity used in a filename(s)" (https://bids-specification--2123.org.readthedocs.build/en/2123/modality-agnostic-files/phenotypic-and-assessment-data.html).

Wowza, good catch! I didn't notice that incorrect set of words there. It currently says:

> (run_id is) A run identifier that corresponds to an existing run-<index> entity used in a filename(s). A chronological run number is used when a measurement tool or assessment described by a tabular file was repeated within a session.

What it should say:

> (run_id is) A run identifier to differentiate multiple runs of the same measurement tool. A chronological run number is used when a measurement tool or assessment described by a tabular file is repeated within a session.

Thanks! I'll edit that.

> For example, I conduct a mood questionnaire 7 times over the course of one session in which multiple tasks (with multiple runs) are conducted.

Does the text edit above of "what it should say" make better sense and resolve the run_id labeling issue?

Relative session acquisition times MAY be listed as durations from
the earliest session (baseline) in days, months, or years
using the `acq_time` column.

## Summary

This appendix described seven guidelines for best tabular phenotypic data.
A short summary table here describes when to use which files.

| File                           | Single session data | Multiple session data |
| :----------------------------- | :------------------ | :-------------------- |
| Participants                   | RECOMMENDED         | RECOMMENDED           |
| Phenotypic and assessment data | RECOMMENDED         | RECOMMENDED           |
| Sessions                       | OPTIONAL            | REQUIRED              |
| Demographics                   | OPTIONAL            | RECOMMENDED           |

## Examples

What follows are a few common use case examples for tabular phenotypic files.

### 1 participant session with both non-tabular and tabular phenotypic data

File tree

```Text
phenotype/
    <measurement_tool_name>.json
    <measurement_tool_name>.tsv
sub-01/anat/
    sub-01_T1w.json
    sub-01_T1w.nii.gz
```

Contents of `phenotype/<measurement_tool_name>.tsv`

> ```Text
> participant_id measurement_1 measurement_2
> sub-01 value1 value2

Author: ```suggestion
```tsv
participant_id	measurement_1	measurement_2
sub-01	value1	value2
```

```

### 1 participant with 2 sessions, where 1 session is only tabular phenotype and the other is only imaging

With only one imaging and one phenotypic session each in this example you might want
to merge both imaging and phenotypic data under one session. But it is more correct to
have separate sessions for the imaging and phenotypic data, especially if
the sessions were collected days, weeks, or months apart. You can denote both sessions
and their acquisition time in the `sessions.tsv` file and have `session_id` `Levels` noted
in the `sessions.json` sidecar. Below are a CORRECT and an INCORRECT example
of prepared data following these guidelines.

#### CORRECT

File tree

```Text
phenotype/
    <measurement_tool_name>.json
    <measurement_tool_name>.tsv
sub-01/ses-MRI/anat/
    sub-01_ses-MRI_T1w.json
    sub-01_ses-MRI_T1w.nii.gz
```

Contents of `phenotype/<measurement_tool_name>.tsv`

> ```Text
> participant_id session_id measurement_1 measurement_2
> sub-01         ses-pheno  value1        value2

Author: ```suggestion
```tsv
participant_id	session_id	measurement_1	measurement_2
sub-01	ses-pheno	value1	value2
```

```

#### INCORRECT

File tree

```Text
phenotype/
    <measurement_tool_name>.json
    <measurement_tool_name>.tsv
sub-01/anat/
    sub-01_T1w.json
    sub-01_T1w.nii.gz
```

Contents of `phenotype/<measurement_tool_name>.tsv`

```Text
participant_id measurement_1 measurement_2
sub-01         value1        value2
```

A session directory **MUST** be present in the participant directory and
the `session_id` column **MUST** be present in `<measurement_tool_name>.tsv` as well.
Sessions must be used consistently for the combination of tabular and
non-tabular phenotypic data.

### 2 participants with a mix of tabular phenotypic data and imaging sessions

File tree

```Text
phenotype/
    <measurement_tool_name>.json
    <measurement_tool_name>.tsv
sub-01/
    ses-MRI1/
        anat/
            sub-01_ses-MRI1_T1w.json
            sub-01_ses-MRI1_T1w.nii.gz
    ses-MRI2/
        anat/
            sub-01_ses-MRI2_T1w.json
            sub-01_ses-MRI2_T1w.nii.gz
sub-02/
    ses-MRI1/
        anat/
            sub-02_ses-MRI1_T1w.json
            sub-02_ses-MRI1_T1w.nii.gz
```

Contents of `phenotype/<measurement_tool_name>.tsv`

> ```Text
> participant_id session_id measurement_1 measurement_2
> sub-01         ses-pheno1 value1        value2
> sub-02         ses-pheno1 value3        value4
> sub-02         ses-pheno2 value5        value6

Author: ```suggestion
```tsv
participant_id	session_id	measurement_1	measurement_2
sub-01	ses-pheno1	value1	value2
sub-02	ses-pheno1	value3	value4
sub-02	ses-pheno2	value5	value6
```

  the same `session_id`, a `run` column MUST be added.

1.  To encode the acquisition time for a measurement tool’s `session_id`,
  add the `session_id` to the sessions file and
  include the OPTIONAL `acq_time` column.

To summarize this guideline as a table:

| **Column name**  | **Requirement** | **Description** |
| :--------------- | :-------------- | :-------------- |
| `participant_id` | REQUIRED        | MUST be the first column in the file.   Note that data for one participant MAY be represented across multiple rows in case of multiple sessions or runs, and therefore the entry in the `participant_id` column will be repeated. |
| `session _id`    | CONDITIONAL ; If sessions are defined in the dataset | A `session_id` column MUST be added to all tabular files in the phenotype directory as soon as multiple sessions are present in the data set regardless of whether those sessions are in the  `phenotype/` data, `sub-<label>/` data, or a combination of the two. |
| `run`            | CONDITIONAL ; If there are multiple runs within any session | A chronological `run` number is used when a measurement tool or assessment described by a tabular file was repeated within a session. |

> | `acq_time`       | OPTIONAL        | If acquisition time is available, the `acq_time` column CAN be used to record the time of acquisition of each row in the tabular file. |

Author: @effigies Is there a macro we can use maybe to clean this table up? How would you recommend making this table more manageable?

Reviewer: https://github.com/bids-standard/bids-specification/blob/ac1148341a5f23990ba599064c56ef4e88d4c80c/src/schema/rules/tabular_data/modality_agnostic.yaml#L2-L19

https://github.com/bids-standard/bids-specification/blob/ac1148341a5f23990ba599064c56ef4e88d4c80c/src/modality-agnostic-files/data-summary-files.md?plain=1#L22-L28

# Tabular phenotypic data guidelines

This appendix is a collection of guidelines and examples
for creating well-organized aggregated tabular phenotypic data.

## Guidelines

These guidelines all apply when the
[`AdditionalValidation` key](../modality-agnostic-files/dataset-description.md#additional-validation)
contains `"Phenotype"` in the `dataset_description.json`.
They are intended to improve the organization and clarity of
tabular phenotypic data like the participants file, sessions file,
and phenotypic and assessment data.

### 1. Aggregate data across sessions

Aggregation refers to the contents of the TSV file. It is REQUIRED
to collect all participant data into one TSV per tabular phenotypic file.

### 2. Always pair tabular data with data dictionaries

Tabular phenotypic data MUST be prepared as one pair of a tabular file
in tab-separated value (TSV) format and a corresponding data dictionary

> in JavaScript Object Notation (JSON) format.

Author: Done.

### 3. Add `MeasurementToolMetadata` to each tabular phenotypic measurement tool

Whenever possible, it is RECOMMENDED to add `MeasurementToolMetadata` to
each `phenotype/<measurement_tool_name>.json` data dictionary.

> This improves reusability and provides clarity about the measurement tool.

Author: Done.

With only one imaging and one phenotypic session each in this example you might want
to merge both imaging and phenotypic data under one session. But it is more correct to
have separate sessions for the imaging and phenotypic data, especially if
the sessions were collected days, weeks, or months apart. You can denote both sessions
and their acquisition time in the `sessions.tsv` file and have `session_id` `Levels` noted
in the `sessions.json` sidecar. Below are a CORRECT and an INCORRECT example
of prepared data following these guidelines.

#### CORRECT

File tree

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example(
   {
   "phenotype": {
      "measurement_tool.json": "",
      "measurement_tool.tsv": "",
      },
   "sub-01": {
      "ses-MRI": {
         "anat": {
            "sub-01_ses-MRI_T1w.json": "",
            "sub-01_ses-MRI_T1w.nii.gz": "",
            }
         }
      }
   }
) }}

Contents of `phenotype/measurement_tool.tsv`

```tsv
participant_id	session_id	measurement_1	measurement_2
sub-01	ses-pheno	value1	value2

> ```

Author: Done.

# Tabular phenotypic data guidelines

This appendix is a collection of guidelines and examples
for curating well-organized tabular phenotypic data.

## Guidelines

These guidelines are intended to improve the organization and clarity of
tabular phenotypic data like the participants file, sessions file,
and phenotypic and assessment data.

> They are recommendations and are by default ignored during validation.

Reviewer 2: Would it be useful to mention that these recommendations are ignored by default only to maintain backward compatibility for previous datasets?

Author: Yes, something like that would be a good add. Adding in next commit.

[`AdditionalValidation` key](../modality-agnostic-files/dataset-description.md#additional-validation)
to contain `"Phenotype"` in the `dataset_description.json`.

### 1. Aggregate data across sessions using `"IndexColumns"`

In multi-session datasets,
aggregate phenotypic and assessment data across all sessions
into one tabular tab-separated value (TSV) file per measurement tool.
In order to aggregate, you MUST use the `"IndexColumns"` list/array field
in the corresponding JavaScript Object Notation (JSON) sidecar file.
There are two examples of this usage [below in this appendix](#examples).
Store each of the TSV and JSON files in the `/phenotype` directory
using the file-naming template `/phenotype/tool-<ToolName>_phenotype.tsv`.
Read the [phenotypic and assessment data section](../modality-agnostic-files/phenotypic-and-assessment-data.md)
for further explanation of how to use `"IndexColumns"`
to aggregate longitudinal or multi-session tabular phenotypic data.

### 2. Always pair tabular data with data dictionaries

Tabular phenotypic data MUST be prepared as one pair of a tabular file
in TSV format and a corresponding data dictionary in JSON format.

> See the [Tabular files section](../common-principles.md#tabular-files) for more information.

Reviewer 3: Should mention this includes participants.tsv

Author: Adding in the next commit.

### 3. Add `MeasurementToolMetadata` to each tabular phenotypic measurement tool

Whenever possible, it is RECOMMENDED to add `MeasurementToolMetadata` to

> each `phenotype/<measurement_tool_name>.json` data dictionary.

Reviewer 2: ```suggestion
each `phenotype/tool-<ToolName>_phenotype.json` data dictionary.
```

According to the file-naming template just above (L26), this should now use `tool-..._phenotype` instead of only `<measurement_tool_name>`

This improves reusability and provides clarity about the measurement tool.

> See [`MeasurementToolMetadata` in the glossary](../glossary.md#measurementtoolmetadata-metadata) for more.

Reviewer 3: Proposal make `MeasurementToolMetadata` required, and the require one of `TermURL` or `Description` to be defined.

Author: Adding to next commit. Thanks for that suggestion!

in each JSON sidecar for multi-session data.
Some common index columns are
`participant_id`, `session_id`, and `run_id`.
Other time, visit, and day based index columns are OPTIONAL.

<!-- This block generates a columns table.
The definitions of these fields can be found in
  src/schema/rules/tabular_data/*.yaml
and a guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_columns_table("modality_agnostic.Phenotypes") }}

Furthermore, if you add a `session_id` index column to any tabular phenotypic data,
you MUST introduce a session directory to the imaging data,
even if only one imaging session has been created.
And vice versa, if imaging data has session directories,
all imaging data and tabular phenotypic data MUST have sessions.

This produces files in which same-participant entries can take up as many rows as needed
according to the smallest unit of acquisition.

### 5. Use a demographics file for multi-session data

If there is more than one session for any one participant, then
it is REQUIRED to provide a demographics file in the `/phenotype` directory

> named as `/phenotype/tool-Demographics_phenotype.tsv`

Reviewer 4: Can the D in `Demographics` be lowercase? Or is there a reasoning behind it being capitalized?  
```suggestion
named as `/phenotype/tool-demographics_phenotype.tsv`
```

Author: That's a large can of de-capitalization worms, but alright. I may add a sentence or note about how people can use lowercase (`tool-lowercase`) or camel case (`tool-camelCase`) or Pascal case (`tool-PascalCase`), just as long as whatever they choose is alphanumeric-only. Just like any other BIDS entity.

            }
         },
      "ses-MRI2": {
         "anat": {
            "sub-01_ses-MRI2_T1w.json": "",
            "sub-01_ses-MRI2_T1w.nii.gz": "",
            }
         }
      },
   "sub-02": {
      "ses-MRI1": {
         "anat": {
            "sub-02_ses-MRI1_T1w.json": "",
            "sub-02_ses-MRI1_T1w.nii.gz": "",
            }
         }
      }
   }
) }}

Contents of `phenotype/tool-Measurements_phenotype.tsv`

```tsv
participant_id	session_id	acq_time	measurement_1	measurement_2
sub-01	ses-MRI1	2001-01-01T11:12:00	value1	value2
sub-01	ses-MRI2	2001-07-01T13:14:00	n/a	n/a
sub-02	ses-MRI1	2001-01-181T15:16:00	value3	value4
sub-02	ses-pheno	2001-02-20T12:05:00	value5	value6
```

### 3 participants with 3 different kinds of sessions among them

The `ses-baseline` session collects an MRI and tabular phenotypic data.

File tree

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example(
   {
   "participants.json": "",
   "participants.tsv": "",
   "phenotype": {
      "tool-Demographics_phenotype.json": "",

>       "tool-Demographics_phenotype.tsv": "",

Reviewer 4: ```suggestion
      "tool-demographics_phenotype.json": "",
      "tool-demographics_phenotype.tsv": "",
```

      "tool-Survey_phenotype.json": "",
      "tool-Survey_phenotype.tsv": "",
      },
   "sub-01": {
      "ses-baseline/": "",
      "ses-followupMRI/": "",
      },
   "sub-02": {
      "ses-baseline/": "",
      },
   "sub-03": {
      "ses-baseline/": "",
      "ses-followupMRI/": "",
      }
   }
) }}

Contents of `participants.tsv`.
Unchanging participant properties belong here.

```tsv
participant_id	sex
sub-01	M
sub-02	F
sub-03	F
```

> Contents of `phenotype/tool-Demographics_phenotype.tsv`.

Reviewer 4: ```suggestion
Contents of `phenotype/tool-demographics_phenotype.tsv`.
```

Participant properties that can change
from session to session belong here especially.

```tsv
participant_id	session_id	acq_time	age	gender	race	household_income
sub-01	ses-baseline	2001-01-01T12:05:00	10	3	4	5
sub-01	ses-followupMRI	2001-07-01T13:33:00	10	3	4	5
sub-01	ses-interview	2002-01-01T11:21:00	11	4	4	6
sub-02	ses-baseline	2001-04-01T11:01:00	9	1	3	3
sub-02	ses-interview	2002-04-01T14:08:00	10	1	7	3
sub-03	ses-baseline	2001-09-01T11:45:00	11	2	10	4
sub-03	ses-followupMRI	2002-03-01T12:17:00	12	5	10	4
```

> Partial contents of `phenotype/tool-Demographics_phenotype.json`.

Reviewer 4: ```suggestion
Contents of `phenotype/tool-demographics_phenotype.tsv`.
Participant properties that can change
from session to session belong here especially.

```tsv
participant_id	session_id	acq_time	age	gender	race	household_income
sub-01	ses-baseline	2001-01-01T12:05:00	10	3	4	5
sub-01	ses-followupMRI	2001-07-01T13:33:00	10	3	4	5
sub-01	ses-interview	2002-01-01T11:21:00	11	4	4	6
sub-02	ses-baseline	2001-04-01T11:01:00	9	1	3	3
sub-02	ses-interview	2002-04-01T14:08:00	10	1	7	3
sub-03	ses-baseline	2001-09-01T11:45:00	11	2	10	4
sub-03	ses-followupMRI	2002-03-01T12:17:00	12	5	10	4
```

Partial contents of `phenotype/tool-demographics_phenotype.json`.
```

> Partial contents of `phenotype/tool-Demographics_phenotype.json`.

Reviewer 4: ```suggestion
Partial contents of `phenotype/tool-demographics_phenotype.json`.
```

Note how the `session_id` `Levels` are clearly described
and how `"IndexColumns"` is present.

```json
{
    "IndexColumns": [
        "participant_id",
        "session_id"
    ],
    "participant_id": {
        "Description": "Participant identifier"
    },
    "session_id": {
        "Description": "Session identifier",
        "Levels": {
            "ses-baseline": "Baseline visit for MRI and assessments",
            "ses-followupMRI": "6-months after baseline MRI follow-up",
            "ses-interview": "1-year after baseline in-person follow-up"
        }
    },
    "acq_time": {
        "Description": "When the data acquisition started"
    }
}
```

> Contents of `phenotype/tool-Survey_phenotype.tsv`.

Reviewer 4: ```suggestion
Contents of `phenotype/tool-survey_phenotype.tsv`.
```

Note how `sub-03` does not have a row for `ses-interview`
because that session was not collected and is absent above

> in the `phenotype/tool-Demographics_phenotype.tsv` file as well.

Reviewer 4: ```suggestion
in the `phenotype/tool-demographics_phenotype.tsv` file as well.
```

The phenotypic and assessment data directory
is for phenotypic measurement instruments collected about the participants
such as questionnaires, surveys, and cognitive assessments.
Create one tabular file for each instrument
in the phenotypic and assessment data directory.

### 6. Record participant properties in the participants file and session properties in the sessions file

Since the same `participant_id` and `session_id` columns can be used
similarly in the participants file and the sessions file,
use the two different files to instead differentiate
properties of participants versus sessions.
Properties of participants MAY include things like
age, sex, race, or household income.
Properties of sessions MAY include things like
acquisition time, measurement device properties,
and indoor or outdoor experimental conditions.

### 7. Use the sessions file at the root-level

If there is more than one session for any one participant, then
it is REQUIRED to provide a sessions file at the dataset root,
even if this file only contains a two-column inventory
of `participant_id` and `session_id`.
The `sessions.tsv` file MUST list all sessions for all subjects across

> imaging and tabular phenotypic data. The `sessions.json` file’s

Reviewer 5: Are we in public review already? here is one:

if you are talking about having top level `sessions.tsv` to now have both `participant_id` and `session_id` index keys, I will keep coming to beg on my knees "PLEASE DO NOT" (edited) until we define it as a generic principle one way or another (e.g. #2273) -- there should be a discussion on that particular aspect **alone** not `subjects.tsv` ad-hoc, and not in BEP mixing in various changes. ATM `sessions.tsv` must have only `session_id` index, as any other `{entity_plural}.tsv`.  Please do not break the few consistencies we have.

Author: We are not in public review yet, but I am happy to receive feedback at all stages.

I see a need in BIDS for allowing flexibility in tabular data aggregating/moving "back" in the directory hierarchy (from subdirectory "leaves" to the top-level dataset "root"). When people interact with this kind of tabular data, they don't want 10's, 100's, or 1,000's of small `sub-<label>/sub-<label>_sessions.tsv` segregated files. They want one big aggregated file.

As a consequence of that: the closer to the dataset root you go, the more columns you will need to describe the aggregated data. For instance, if you have 1,000's of small `sub-<label>/sub-<label>_sessions.tsv` segregated files and you want one big aggregated file called `sessions.tsv` instead, aggregating up requires taking out the `sub-<label>`. So you end up with the need for that `participant_id` column in the `sessions.tsv` file.

I used to think this was relevant for ALL tabular files to be able to move back in the directory hierarchy, but I later realized that things like physio or other large files would be a real mess in a much larger aggregated file. I could maybe still make the case for scans.tsv or samples.tsv though, but that's a very different story...

I would almost like to say, "'Large' tabular files should stay as-is and 'small' tabular files should allow aggregation."

Reviewer 5: sorry -- I am not following the argument here as nothing in what you said requires breaking an established  `.tsv` file(s) convention as it can be easily adopted in a new .tsv file as e.g. I suggest in #2273.

It is RECOMMENDED to store acquisition time[<sup>2</sup>](#footnotes)
for tabular phenotypic data and store the time of acquisition of each row
inside a column named `acq_time` in the sessions file.
This is consistent with how acquisition time is recorded for MRI data
and other time-sensitive measurements (for example systolic blood pressure).

## Summary

This appendix described guidelines for best tabular phenotypic data.
In summary, it is RECOMMENDED to always use the participants file
and separate files by measurement instrument in
the phenotypic and assessment data directory,
since they each collect different information.
If you use sessions, then the sessions file is also RECOMMENDED.

## Examples

What follows are a few common use case examples for tabular phenotypic files.

### 1 participant session with both non-tabular and tabular phenotypic data

File tree

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example(
   {
   "phenotype": {
      "measurement_tool.json": "",

>       "measurement_tool.tsv": "",

Reviewer 6: This naming convention with the underscore breaks common BIDS rules: is this meant to be a proposal for an '_tool' suffix?

      "measurement_tool.tsv": "",
      },
   "sub-01": {
      "anat": {
         "sub-01_T1w.json": "",
         "sub-01_T1w.nii.gz": "",
         }
      }
   }
) }}

Contents of `phenotype/measurement_tool.tsv`

```tsv
participant_id	measurement_1	measurement_2
sub-01	value1	value2
```

A session directory **MUST** be present in the participant directory and
the `session_id` column **MUST** be present in `phenotype/measurement_tool.tsv` as well.
Sessions must be used consistently for the combination of tabular and
non-tabular phenotypic data.

### 2 participants with a mix of tabular phenotypic data and imaging sessions

In this example, participants acquired both
a phenotypic measurement tool and an MRI during `ses-MRI1`.
`sub-01` has a `ses-MRI2` with no phenotypic measurement tool acquired
and `sub-02` has a `ses-pheno` where no MRI was acquired.

File tree

<!-- This block generates a file tree.
A guide for using macros can be found at
 https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
-->
{{ MACROS___make_filetree_example(
   {
   "sessions.json": "",
   "sessions.tsv": "",
   "phenotype": {
      "measurement_tool.json": "",

>       "measurement_tool.tsv": "",

Reviewer 5: yikes -- that will be the literal filename like `measurement_tool.tsv` in there??? like the suffix `_tool` and some `measurement` *thing*?

Author: Yes, this is what the `phenotype/` directory already permits and the validator validates pre-BEP036. It is also how the 40 (out of 1,548) datasets with a `phenotype/` directory on OpenNeuro, as of 3/30/2026, curated their tabular phenotypic data. The filenames in the `phenotype/` directory can be named alphanumerically (including hyphens and plus signs, I believe), but I think the practical way to make sure people continue sharing their tabular phenotypic data is to leave the file-naming of phenotype files as-is.

Names so far are usually acronyms or short descriptions or both. Some examples:

- `phenotype/demographics.tsv`
- `phenotype/whodas.tsv`
- `phenotype/nih_toolbox.tsv`
- `phenotype/ARI_parent.tsv`
- `phenotype/wisc-iii.tsv`

Reviewer 5: > but I think the practical way to make sure people continue sharing their tabular phenotypic data is to leave the file-naming of phenotype files as-is.

by no means I want to restrict people from sharing their data any way they want in whatever file-names they have.  It is only when we talk about standardizing file names like BIDS does, that's where I have an "opinion" or otherwise we just would have ended-up with something like `data/my_favorite1.nii.gz`  and `data/notsogoodone2.nii.gz`

Reviewer 6: Agree with @yarikoptic that this breaks BIDS search principles for use of underscores.

      },
   "sub-01": {
      "ses-MRI1": {
         "anat": {
            "sub-01_ses-MRI1_T1w.json": "",
            "sub-01_ses-MRI1_T1w.nii.gz": "",
            }
         },
      "ses-MRI2": {
         "anat": {
            "sub-01_ses-MRI2_T1w.json": "",
            "sub-01_ses-MRI2_T1w.nii.gz": "",
            }
         }
      },
   "sub-02": {
      "ses-MRI1": {
         "anat": {
            "sub-02_ses-MRI1_T1w.json": "",
            "sub-02_ses-MRI1_T1w.nii.gz": "",
            }
         }
      }
   }
) }}

Contents of `sessions.tsv`

```tsv
participant_id	session_id	acq_time

> sub-01	ses-MRI1	2001-01-01T11:12:00

Reviewer 5: please please please **no**!!! what common principle such file construction follows?  it introduces inconsistencies!  If you want to introduce some new principle, let's formalize it. PLEASE PLEASE PLEASE lets discuss and do that, e.g. in https://github.com/bids-standard/bids-specification/issues/2273 or as I invited you there -- **please formulate an alternative but generic approach!**

You can keep resolving my comments but I am sorry -- I will be coming back, so we better do talk and resolve this "conflict"!

Author: Please do keep coming back. We want the voices of those that agree AND those that have differing opinions to see whether we're satisfying 80% of the most commonly used data cases or not (see [BIDS mission statement](https://bids.neuroimaging.io/collaboration/governance.html#b-bids-mission-statement) foundational principle 2). I will try to make an "alternative, but generic approach" and tag you on it referencing #2273, etc. Hopefully we can come to some agreement soon.

# Tabular phenotypic data guidelines

This appendix is a collection of guidelines and examples
for curating well-organized tabular phenotypic data.

## Guidelines

These guidelines are intended to improve the organization and clarity of
tabular phenotypic data like the participants file, sessions file,
and phenotypic and assessment data.

They are recommendations and are by default ignored during validation.
You can make them mandatory during validation by setting the
[`AdditionalValidation` key](../modality-agnostic-files/dataset-description.md#additional-validation)
contains `"Phenotype"` in the `dataset_description.json`.

### 1. Aggregate data across sessions

Aggregate participant information across all sessions into one tabular TSV file per
measurement or phenotypic assessment and store this file in the `/phenotype` directory.

> Demographic information is a special case and  MUST be aggregated

Author: I think a growing list of specific field names considered common demographics would be great: `sex`, `age`, `gender`, `race`, and `income` for starters. Though perhaps that should be a validator WARNING and not an ERROR, so I will de-escalate that "MUST" to a "SHOULD".

Your comment also raises to me the thought of "does the validator check for the presence of duplicate-named columns across tabular data"? While I don't think it's a good idea to duplicate column names, it might happen sometimes and should raise a WARNING to encourage people to de-duplicate.

> Demographic information is a special case and  MUST be aggregated

Reviewer: > "does the validator check for the presence of duplicate-named columns across tabular data"?

It does not.

> it might happen sometimes and should raise a WARNING to encourage people to de-duplicate.

Our problem remains that we emit so many warnings that people simply stop reading them.

in the `participants.tsv` file at the root level of the dataset.

> It is RECOMMENDED to use the `age` column in the `participants.tsv` file

Reviewer 3: Theoretically we could validate the appropriate age being used in each session based on the relative acq_times if present but I don't think its worth the effort. Maybe monotonically increasing age like schema.rules.checks.mri.VolumeTimingNotMonotonicallyIncreasing would be a compromise?

Author: I agree it's not worth the effort. That and we can't rely on the age monotonically increasing as some sessions may be close enough to not affect the reported age.

to record participant age at every session in longitudinal or multi-session data sets.

### 2. Always pair tabular data with data dictionaries

Tabular phenotypic data MUST be prepared as one pair of a tabular file
in tab-separated value (TSV) format and a corresponding data dictionary
in JavaScript Object Notation (JSON) format.
See the [Tabular files section](../common-principles.md#tabular-files) for more information.

### 3. Add `MeasurementToolMetadata` to each tabular phenotypic measurement tool

> Whenever possible, it is RECOMMENDED to add `MeasurementToolMetadata` to

Author: @rwblair Do you think we should do away with the "Measurement Tool" language everywhere and replace it with something like "tabular data, which may be aggregated or collected from multiple measurements"?

each `phenotype/<measurement_tool_name>.json` data dictionary.
This improves reusability and provides clarity about the measurement tool.
See [`MeasurementToolMetadata` in the glossary](../glossary.md#measurementtoolmetadata-metadata) for more.

### 4. Ensure minimal annotation for phenotypic and assessment data

In phenotypic and assessment data, each measurement tool SHOULD have an independent
aggregated data TSV file in which the user collects all subjects, sessions,
and/or runs of data as one entry per row (with a row defined by
the smallest unit of acquisition). In other words:

-   Each row MUST start with `participant_id`.

-   Each TSV file MUST contain a `session_id` column when
    multiple [sessions](../glossary.md#session-entities)[<sup>1</sup>](#footnotes) are present
    in the data set regardless of whether those sessions are in
    the `phenotype/` data, `sub-<label>/` data, or a combination of the two.

-   If more than one of the same measurement tool is acquired within

>     the same `session_id`, a `run_id` column MUST be added.

Author: Love the edit! Implementing it now. Thanks!

-   Encoding the acquisition time for a measurement tool’s `session_id`,
    is RECOMMENDED. This information MUST be stored in the `sessions.tsv`

>     file at the root level of the dataset in the `acq_time` column.

Reviewer 3: @effigies mentioned this as "This is logically equivalent to "the acq_time column MUST NOT appear in a phenotype TSV file", but it takes some thinking about to get there. The spec should just say that."

I agree with the explicit "MUST NOT", But it also goes a step further in enforcing a root level sessions.tsv.
